synthtool/metadata.py

```diff
@@ -238,3 +242,52 @@ def _clear_local_paths(metadata):
         if source.HasField("git"):
             git_source = source.git
             git_source.ClearField("local_path")
+
+
+def _add_self_git_source():
+    """Adds current working directory as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    # Use the repository's root directory name as the name.
+    return _add_git_source_from_directory(".", os.getcwd())
+
+
+def _add_git_source_from_directory(name: str, dir_path: str):
+    """Adds the git repo containing the directory as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "status"], universal_newlines=True
+    )
+    if completed_process.returncode:
+        log.warning("%s is not directory in a git repo.", dir_path)
+        return 0
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "remote", "get-url", "origin"],
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    url = completed_process.stdout.strip()
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "log", "--no-decorate", "-1", "--pretty=format:%H"],
```

Comment:
might warrant a comment that `-1` actually returns the latest `sha`, or you could just call the variable `latest_sha`?

Reply:
Done.